spec/lib/logatron/logatron_spec.rb

```diff
@@ -31,4 +31,50 @@ module Logatron
       end
     end
   end
+
+  describe '::request_info' do
+    it 'returns a hash of the request-level parameters for serialization' do
+      Logatron.msg_id = '12345'
+      Logatron.site = 'cite'
+      info = Logatron.request_info
+      expect(info).to be_a Hash
+      expect(info[:message_id]).to eql '12345'
+      expect(info[:site]).to eql 'cite'
+    end
+  end
+
+  describe '::parse_request_info' do
+    it 'sets the message id' do
+      Logatron.parse_request_info({'message_id' => '12345'})
+      expect(Logatron.msg_id).to eql '12345'
+
+      Logatron.parse_request_info({message_id: '12345'})
+      expect(Logatron.msg_id).to eql '12345'
+    end
+    it 'sets the site' do
+      Logatron.parse_request_info({'site' => 'medtoxlabcorp'})
+      expect(Logatron.site).to eql 'medtoxlabcorp'
+
+      Logatron.parse_request_info({site: 'medtoxlabcorp'})
+      expect(Logatron.site).to eql 'medtoxlabcorp'
+    end
+    context 'when passed nil values' do
+      it 'leaves the site and message id in their current states' do
+        Logatron.msg_id = 'abcde'
+        Logatron.site = 'corplabtoxmed'
+        Logatron.parse_request_info({site: nil, message_id: nil})
+        expect(Logatron.msg_id).to eql 'abcde'
+        expect(Logatron.site).to eql 'corplabtoxmed'
+      end
+    end
+    context 'when not passed a hash' do
```

Comment:
Technically, this is testing `nil` only. There are many other types of objects that are not hashes; those would likely result in a `TypeError` arising from `#[]`.